crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -1182,8 +1188,19 @@ mod private {
             func: F,
         ) -> Result<(O, Vec<VectorDiff<TimelineEvent>>), EventCacheError> {
             let output = func(&mut self.events);
+
             self.propagate_changes().await?;
+
+            // If we've never waited for an initial previous-batch token, and we now have at
```

Comment:
Why is it here? I feel like this is not the correct place, but it's a gut feeling.

Reply:
Thanks for the explanation. Makes sense to me.